controllers/components/kserve/kserve_controller.go

```diff
@@ -151,6 +151,7 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 			deploy.WithCache(),
 		)).
 		WithAction(security.NewUpdatePodSecurityRoleBindingAction(serviceAccounts)).
+		WithAction(patchOwnerReference).
```

Comment:
we also need a finalizer action to remove the owner reference, as alternative we can remove the owner reference and let the DSC to remove the model controller instance to make things simpler for the time being

Reply:
will this be enough for now? 
https://github.com/opendatahub-io/opendatahub-operator/blob/ed85eab56aa1ade3a2581706037917098099437d/controllers/components/kserve/kserve_controller_actions.go#L285

Reply:
maybe just better remove the owner reference at all at this stage. 

the point I'm trying to make is that, if you delete the KServe resource, then it should also remove the owner reference from the model controller, to ensure it can be garbaged collected by k8s but since we are exciting deleting the model controller CR,  then setting the owner reference does not make much sense.